### ISSUE TITLE
Texture lost after converting ColorVisuals to TextureVisuals

### ISSUE THREAD

**Issue by Steve-Tod** [mikedh/trimesh#1289]
Hi there, I'm trying to color a mesh (no texture) and convert the color to texture with `to_texture` method of `ColorVisuals`. However, after this conversion, the output mesh shows no texture. Here are my scripts:
```
mesh = trimesh.load('path/to/mesh')
# color to red
point = np.array([0, 0, 0.06]).reshape(1, 3)
dist = np.sqrt(np.sum((mesh.triangles_center - point) ** 2, axis=1))
mask = dist < 0.07
mesh.visual.face_colors[mask] = np.array([255, 0, 0, 255], dtype=np.uint8)
# conversion
new_mesh = mesh.copy()
new_mesh.visual = mesh.visual.to_texture()
```

This is the result of `mesh.show()`:
![image](https://user-images.githubusercontent.com/26958167/126024482-27b0f32c-555a-43bc-bd25-4199e68d460d.png)

And this is the result of `new_mesh.show()`:
![image](https://user-images.githubusercontent.com/26958167/126024479-8a0e0a4f-c03a-441a-8c6c-74decb489f99.png)

Do you know why is this happening? The colors are lost after converting. Thank you!

**Comment by ttsesm** [comment]
@Steve-Tod did you manage to resolve this issue?

@mikedh is this the correct way to pass vertex color to texture?

**Comment by ttsesm** [comment]
Ok it seems that the functionality is not implemented completely, since according to functions [`from_color()`](https://github.com/mikedh/trimesh/blob/4aea783457c06ebe5a457a4da070dd22668db923/trimesh/visual/material.py#L652) and [`apply_material()`](https://github.com/mikedh/trimesh/blob/4aea783457c06ebe5a457a4da070dd22668db923/trimesh/visual/material.py#L603) it assigns only one color to the texture map.

**Comment by Steve-Tod** [comment]
Hmm, I see, so I need to rewrite the `from_color()` and `apply_material()` to correctly create a material and a uv mapping, is that right?

**Comment by ttsesm** [comment]
@Steve-Tod yes if you notice on the code in the `from_color()` function there is a `TODO` task. Apparently the tiling/mapping is of the colors is not implemented. I am not quite sure what @mikedh does with the [grouping](https://github.com/mikedh/trimesh/blob/4aea783457c06ebe5a457a4da070dd22668db923/trimesh/visual/material.py#L650) but if I get it correctly he retrieves all the different colors which then they have be mapped to the UV coordinates inside the `apply_material()`.

**Comment by Steve-Tod** [comment]
Got it, thanks!

**Comment by ttsesm** [comment]
@Steve-Tod why did you close the issue? I think you should leave it open until the code is fixed.

**Comment by Steve-Tod** [comment]
Oh, I thought I need to solve it by myself and I kinda know where to look at first based on your responses, so I closed the issue. It would be great if there's an official fix to this!

**Comment by ttsesm** [comment]
Well you can work on it on yourself and if you fix it you could make a pr or update this thread. If I get some time, I might have a look as well.
